### ISSUE TITLE
`Float64` `run_speedy` doesn't output correctly

### ISSUE THREAD

**Issue by rejuvyesh** [milankl/SpeedyWeather.jl#102]
```julia
julia> run_speedy(Float64; n_days=20, model=:shallowwater, output=true, trunc=62, )
ERROR: MethodError: no method matching scale_coslat⁻¹!(::SubArray{Float32, 2, Array{Float32, 3}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}, ::Geometry{Float64})
Closest candidates are:
  scale_coslat⁻¹!(::AbstractMatrix{NF}, ::Geometry{NF}) where NF at ~/.julia/packages/SpeedyWeather/UDuyp/src/spectral_gradients.jl:266
  scale_coslat⁻¹!(::AbstractArray{NF, 3}, ::Any...) where NF at ~/.julia/packages/SpeedyWeather/UDuyp/src/distributed_vertical.jl:5
```
Likely because they are being converted into `Float32` while saving but before scaling.

https://github.com/milankl/SpeedyWeather.jl/blob/1881d47ab3f28a85e433ccaaf5fc7218e4792b8d/src/output.jl#L173-L181

**Comment by milankl** [comment]
Thanks for pointing this out. I’d suggest that we need to implement a corresponding, less specific, method that doesn’t require the array and the Geometry instance to have the same number format NF. In that case we could scale by converting explicitly every element of the coslat^n vectors to Float32. I suggest the explicit conversion as to not run into troubles when NF hasn’t a type promotion to Float32 defined. Given this is not performance critical code I prefer that solution over also precomputing Float32 versions of the coslat^n vectors in Geometry

**Comment by milankl** [comment]
Addressed with #103
